packages/gator-permissions-snap/src/core/permissionRequestLifecycleOrchestrator.ts

```diff
@@ -244,14 +248,11 @@ export class PermissionRequestLifecycleOrchestrator {
       isAdjustmentAllowed,
     };
 
-    const [address, accountMetadata] = await Promise.all([
-      this.#accountController.getAccountAddress({
-        chainId,
-      }),
-      this.#accountController.getAccountMetadata({
-        chainId,
-      }),
-    ]);
+    const { address } = grantedPermissionRequest;
+    // todo: it would be nice if this was concretely typed
+    if (address === undefined) {
```

Comment:
I think it would be better to check with 

```
if(!address)
```

So we also check for null or empty string